errorhandler/error-handler.go

```diff
@@ -0,0 +1,39 @@
+package errorhandler
+
+import (
+	"log"
+	"net/http"
+)
+
+type ErrorHandler interface {
+	Handle(err error, writer http.ResponseWriter)
+}
+
+type CommentErrorHandler struct{}
+
+type PostErrorHandler struct{}
+
+type PostWithCommentsErrorHandler struct{}
```

Comment:
when doing DI, we should hide implementation struct and only expose inteface and constructor